src/js/extensions/ccf/gov.cpp

```diff
@@ -27,34 +27,103 @@ namespace ccf::js::extensions
           ctx, "Passed %d arguments, but expected 1", argc);
       }
 
-      auto constitution = jsctx.to_str(argv[0]);
-      if (!JS_IsString(argv[0]) || !constitution.has_value())
+      auto arg = jsctx.wrap(argv[0]);
+      if (!arg.is_str())
       {
         return JS_ThrowTypeError(ctx, "constitution is not a string");
       }
 
-      if (constitution->size() == 0)
+      auto constitution = jsctx.to_str(arg);
+      if (!constitution.has_value())
+      {
+        return JS_ThrowTypeError(ctx, "constitution is not a string");
+      }
+
+      if (constitution->empty())
       {
         return JS_ThrowTypeError(ctx, "constitution is empty");
       }
 
-      const auto path = "proposed constitution";
+      const std::string path("proposed constitution");
+
+      struct ArgsSpec
+      {
+        std::vector<std::string> required_args;
```

Comment:
Odd to initialise one but not the other?

Reply:
Fine, clang-tidy agrees with you that this is redundant-affecting-readability. I submit, under protest.

Reply:
Time and tidy wait for no man.